main.ts

```diff
@@ -164,10 +220,44 @@ export default class ShareAsGistPlugin extends Plugin {
       }),
     });
 
+    this.addTimeoutUpdater();
+
     // This adds a settings tab so the user can configure various aspects of the plugin
     this.addSettingTab(new ShareAsGistSettingTab(this.app, this));
   }
 
+  addTimeoutUpdater() {
+    let timeout: number | null = null;
+    let previousContent = '';
+
+    this.app.workspace.on('editor-change', async (editor, _info) => {
```

Comment:
I read the [Obsidian release notes](https://obsidian.md/changelog/2021-10-06-desktop-v0.12.17/) where this event was released, and it says the following:

> This event happens fairly frequently so please be mindful of performance when using it.

I wonder if need to be concerned about this from a performance perspective. We might want to debounce this function so it doesn't get called too often.